### PR TITLE
Fish tanks 3.0

### PR DESCRIPTION
**Pull request by jakmak6**
remake of https://github.com/vgstation-coders/vgstation13/pull/13992

Ports paradises fish tanks and sushi recipes.

Here is a webm of it in action

https://puu.sh/tJkcW/16db19a379.webm

You can make the boards for the fish tank and large fish tank via solder or order them from cargo

The eggs and fish care tools are bough from cargo in the aqua culture crate and the exotic fish eggs crate.

To add fish in just insert water and fish eggs to the tank and use the fish egg scoop to harvest any eggs that the fish lay. Re add the eggs to get more fish and use the net to pull fish out of the tank. Use the fish brush to clean the gunk off the tank to ensure happy and healthy fish.

Once harvested salmon and catfish can be gutted using any knife as so they could be used for making fish related recipes

here are the new sprites for the tanks courtesy of cptwad.

https://puu.sh/tZUFw/514a26d7a6.png

Other than the new sprites the code for the fishtanks and fixing sharkteeth code hasnt been changed

### CODE REVIEW

File: code/datums/supplypacks.dm [hunk 1]
```diff
@@ -1737,6 +1737,49 @@ var/list/all_supply_groups = list("Supplies","Clothing","Security","Hospitality"
 	access = access_hydroponics
 	group = "Hydroponics"
 
+/datum/supply_packs/aquaculture // fish n shit
+	name = "Aquaculture Supply"
```
Comment (DamianX) [inline code/datums/supplypacks.dm]: supply vs Supply

File: code/datums/supplypacks.dm [hunk 2]
```diff
@@ -1737,6 +1737,49 @@ var/list/all_supply_groups = list("Supplies","Clothing","Security","Hospitality"
...
+					/obj/item/weapon/circuitboard/fishtank,
+					/obj/item/weapon/circuitboard/fishtank,
+					)
+	cost = 30
+	containertype = /obj/structure/closet/crate/hydroponics
+	containername = "aquaculture crate"
+	group = "Hydroponics"
+
+/datum/supply_packs/exoticfish // weird fish eggs n shit
+	name = "Exotic Fish"
```
Comment (DamianX) [inline code/datums/supplypacks.dm]: fish vs Fish

**Comment by yclatious** [comment]
Are shark teeth compatible with the mayan weapon? Itd be neat if they were

**Comment by Kurfursten** [comment]
Did you fix all that stuff from last time?

**Comment by jakmak6** [comment]
probably not

**Comment by Shadowmech88** [comment]
Shark teeth *could* be compatible with the macuahuitl if they were children of glass shards.

Or, I guess, if someone specifically made the macuahuitl able to take them, but that'd be more work.

**Comment by Kurfursten** [comment]
As mentioned before though, that would mean they wouldn't have teeth behaviors

**Comment by Shadowmech88** [comment]
Oh, right.

I suppose someone could change the macuahuitl to accept teeth in general, which would be more convenient than shark teeth only.

**Comment by jakmak** [comment]
Thing is they were originally chidren of shards but I changed it on kurfs request. I can change it back if you want

**Comment by Shadowmech88** [comment]
I don't mind one way or the other.

**Comment by Kurfursten** [comment]
Tooth macuahuitls should be their own PR so I can finally make xenotooth weapons

**Comment by Riekland** [comment]
My position remains that they shouldn't be put in until they stop looking like shit
what no _I'm_ not gonna sprite it what do you take me for a non piece of shit

**Comment by BlueNexus** [comment]
Sprites are still terrible,, and worse than before.

**Comment by ultraplen** [comment]
Why are the two tanks drawn in different perspectives ?

File: code/modules/fish/fish_tank.dm [hunk 3]
```diff
@@ -0,0 +1,692 @@
...
+
+	return
+
+//////////////////////////////
+//		PROCESS PROC		//
+//////////////////////////////
+
+//Stops atmos from passing wall tanks, since they are effectively full-windows.
+///obj/machinery/fishtank/wall/CanAtmosPass(var/turf/T) paradise code yo
+	return 0
```
Comment (Kurfursten) [inline code/modules/fish/fish_tank.dm:175]: aaaaa

**Comment by Kurfursten** [comment]
@ultraplen In the original version, the large tank is supposed to work as a wall and block everything even air

**Comment by ultraplen** [comment]
Okay. If it no longer blocks air and presumably only has a larger volume, do we really need both aquariums? The larger one could be scrapped, since it has the worse sprite.

**Comment by gbasood** [comment]
You need to actually figure out git at some point dude
This is getting ridiculous

**Comment by jakmak6** [comment]
sood i remade it because the old PR was before https://github.com/vgstation-coders/vgstation13/pull/14380

**Comment by gbasood** [comment]
okay but account # 6 tho

**Comment by Riekland** [comment]
Okay so doing some looking back for other autism I found this comment by our lord and savior
https://github.com/vgstation-coders/vgstation13/issues/7946#issuecomment-177064712
I think that the two fishtank groups while not perfect are at least better than the ones in this PR
ps @Zth-- where's the fish tanks you promised HUH

**Comment by yclatious** [comment]
Never ever

**Comment by FalseIncarnate** [comment]
Hey there, I'm the guy who coded the fishtanks on Paradise initially, cool to see someone porting it elsewhere. If you wouldn't mind dropping a link back to the original PR somewhere in your description, I'd really appreciate that so that the coders and spriters who contributed to them can get their fair credit.

As Kuru stated, the large tanks are called "Wall Aquariums" because they take up the same space as a wall, and (at least for us) block atmos like a wall. They also have a higher volume both in terms of water and fish capacity, as well as a bit more health than the smaller aquarium. The perspective difference was mostly an issue of available space and trying to avoid any weird gaps if you use them as part of a wall, though I admit it definitely is a bit jarring in comparison to the (much better) perspective of the smaller tank.

Shark teeth were children of glass shards so that you could load them into pneumatic cannons for shrapnel cannons or leave them on the ground to shred the feet of the shoeless crew, though that was also mostly laziness on my part. Changing that is honestly fine (I might backport that change to be frank).

I would make sure you grab the most recent changes to fishtanks, as I cleaned up the code and made some tweaks to fish breeding. Not absolutely necessary, but if you plan to keep up with our fish, you'll find it easier to do if you have that stuff already when more is added.

**Comment by Intigracy** [comment]
conflicts

**Comment by ghost** [comment]
fix it!

**Comment by gbasood** [comment]
ftfy

File: code/modules/fish/fish_items.dm [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+	desc = "A tiny net to capture fish with. It's a death sentence!"
+	icon = 'icons/obj/fish_items.dmi'
+	icon_state = "net"
+	slot_flags = SLOT_BELT
+	throwforce = 0
+	w_class = 2
+	throw_speed = 3
+	throw_range = 7
+
+	suicide_act(mob/user)			//"A tiny net is a death sentence: it's a net and it's tiny!" https://www.youtube.com/watch?v=FCI9Y4VGCVw
```
Comment (ghost) [inline code/modules/fish/fish_items.dm:41]: partial pathing is a no-no

File: code/modules/fish/fish_items.dm [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+	icon = 'icons/obj/fish_items.dmi'
+	icon_state = "brush"
+	slot_flags = SLOT_BELT
+	throwforce = 0
+	w_class = 2
+	throw_speed = 3
+	throw_range = 7
+	attack_verb = list("scrubbed", "brushed", "scraped")
+
+	suicide_act(mob/user)
```
Comment (ghost) [inline code/modules/fish/fish_items.dm:67]: partial pathing!

File: code/modules/fish/fish_items.dm [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+//////////////////////////////////////////////
+
+/obj/item/weapon/reagent_containers/food/snacks/shrimp
+	name = "shrimp"
+	desc = "A single raw shrimp."
+	icon = 'icons/obj/seafood.dmi'
+	icon_state = "shrimp_raw"
+	filling_color = "#FF1C1C"
+
+	New()
```
Comment (ghost) [inline code/modules/fish/fish_items.dm:82]: Partial pathing

File: code/modules/fish/fish_items.dm [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+		src.bitesize = 1
+
+/obj/item/weapon/reagent_containers/food/snacks/feederfish
+	name = "feeder fish"
+	desc = "A tiny feeder fish. Sure doesn't look very filling..."
+	icon = 'icons/obj/seafood.dmi'
+	icon_state = "feederfish"
+	filling_color = "#FF1C1C"
+
+	New()
```
Comment (ghost) [inline code/modules/fish/fish_items.dm:95]: yet more partial pathing

**Review by ghost — CHANGES_REQUESTED** [review]
partial pathing needs changing

**Comment by Shadowmech88** [comment]
Was this ready to go aside from conflicts and partial pathing?

**Comment by Intigracy** [comment]
Seems like it

**Pull request closed without merging**